templates/testdata/cluster-with-failure-domain.yaml

```diff
@@ -0,0 +1,79 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  labels:
+    ccm: nutanix
+    cluster.x-k8s.io/cluster-name: cluster-with-failure-domains
+  name: cluster-with-failure-domains
+spec:
+  topology:
+    class: nutanix-quick-start
+    controlPlane:
+      metadata: {}
+      replicas: 1
+    variables:
+      - name: sshKey
+        value: ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIMe61GqA9gqeX3zDCiwuU8zEDt3ckLnfVm8ZxN7UuFyL user@host
+      - name: controlPlaneEndpoint
+        value:
+          IP: 1.2.3.4
+          port: 6443
+      - name: prismCentralEndpoint
+        value:
+          additionalTrustBundle: cluster-with-failure-domains-pc-trusted-ca-bundle
+          address: prismcentral.fake
+          credentialSecret: nutanix-quick-start-pc-creds
+          insecure: false
+          port: 9440
+      - name: controlPlaneMachineDetails
+        value:
+          bootType: legacy
+          clusterName: fake
+          imageName: ubuntu-2204-kube-v1.29.2.qcow2
+          memorySize: 4Gi
+          subnetName: fake-subnet
+          systemDiskSize: 40Gi
+          vcpuSockets: 2
+          vcpusPerSocket: 1
+      - name: workerMachineDetails
+        value:
+          bootType: legacy
+          clusterName: fake
+          imageName: ubuntu-2204-kube-v1.29.2.qcow2
+          memorySize: 4Gi
+          subnetName: fake-subnet
+          systemDiskSize: 40Gi
+          vcpuSockets: 2
+          vcpusPerSocket: 1
+      - name: failureDomains
+        value:
+          - cluster:
+              name: cluster1
+              type: name
+            controlPlane: true
+            name: cluster1
+            subnets:
+              - name: fake-subnet
+                type: name
+          - cluster:
+              name: cluster2
+              type: name
```

Comment:
we should add a test with uuid cluster as well. same for subnet

Reply:
will update the existing test to have one cluster and subnet defined w/ UUIDs